addon/validators/presence.js

```diff
@@ -1,12 +1,20 @@
 import buildMessage from 'ember-changeset-validations/utils/validation-errors';
 import { validate } from 'ember-validators';
 
-export default function validatePresence(options) {
+export default function validatePresence(options = {}) {
+  let targets
   if (typeof options === 'boolean') {
     options = { presence: options };
+  } else {
+    targets = typeof options.on === 'string' ? [ options.on ] : options.on;
+    delete options.on;
   }
 
-  return (key, value) => {
+  return (key, value, oldValue, changes, content) => {
+    if (targets && !targets.some((target) => changes[target] || ( changes[target] === undefined && content[target]))) {
+      return true;
+    }
```

Comment:
can you explain the necessity of this change?  Not that it is confusing or I don't think it is needed, but just a short description to help me understand!

Reply:
The idea is if you want something to `validatePresence`, but only when one or more values or present in the form.  Like making sure someone enters their password if they're doing something involved.  Or make sure someone's address has their state and zip entered as well.

This option bails if the changes none of the targets exist in the changes or, if they don't exist in the changes, exist in the model.  Otherwise, it works just like normal validate presence.